src/functions/op.js

```diff
@@ -90,38 +90,58 @@ const htmlBuilder = (group, options) => {
         // in a new span so it is an inline, and works.
         base = buildCommon.makeSpan([], [base]);
 
-        let supm;
-        let supKern;
-        let subm = {height: 0, depth: 0};  // Make flow happy
-        let subKern;
-        let newOptions;
+        let sub;
+        let sup;
         // We manually have to handle the superscripts and subscripts. This,
         // aside from the kern calculations, is copied from supsub.
         if (supGroup) {
-            newOptions = options.havingStyle(style.sup());
-            supm = html.buildGroup(supGroup, newOptions, options);
+            const elem = html.buildGroup(
+                supGroup, options.havingStyle(style.sup()), options);
 
-            supKern = Math.max(
-                options.fontMetrics().bigOpSpacing1,
-                options.fontMetrics().bigOpSpacing3 - supm.depth);
+            sup = {
```

Comment:
I coupled the `elem` (originally `subm`, `supm`) variable with the corresponding `kern` (originally `subKern`, `supKern`) so that type refinement (`if` checks) below could guarantee when both these values were set and usable.